src/Attributes/src/Internal/Instantiator/Factory.php

```diff
@@ -25,13 +27,19 @@ class Factory implements InstantiatorInterface
      */
     private $named;
 
+    /**
+     * @var AnnotationReader
+     */
+    private $annotationReader;
+
     /**
      * Factory constructor.
      */
     public function __construct()
     {
         $this->doctrine = new DoctrineInstantiator();
         $this->named = new NamedArgumentsInstantiator();
+        $this->annotationReader = new AnnotationReader();
```

Comment:
An annotation reader cannot be instantiated in internal classes because this is an optional dependency. The reader type must be defined by the user himself.